### DISCUSSION TITLE
mcrypt library missing in php installation

### DISCUSSION THREAD

**Discussion by bharatkumarp** [shivammathur/homebrew-php#892] [Q&A / Help] [answered]
Hi, 

I installed php5.6 using command "brew install php@5.6". After installation, when I run "make composer" in our php source, I see error like below.

Error info:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - onelogin/php-saml 2.11.0 requires ext-mcrypt * -> the requested PHP extension mcrypt is missing from your system.
    - onelogin/php-saml 2.11.0 requires ext-mcrypt * -> the requested PHP extension mcrypt is missing from your system.
    - Installation request for onelogin/php-saml 2.11.0 -> satisfiable by onelogin/php-saml[2.11.0].
```
    - 
**Here is the version in my Mac(mcrypt not present in path):**
```
(base) BPERUMAL-M-50R9:website bperumal$ php -v
PHP 5.6.40 (cli) (built: Apr  6 2022 05:39:05)
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
(base) BPERUMAL-M-50R9:website bperumal$
```
My colleague does not have this problem. I checked his Php version. 
**My colleague has php version as php5-5.6.36-20180703-123509(mcrypt libarary exist)**

I did run the command `"brew reinstall $(brew deps shivammathur/php/php@5.6) shivammathur/php/php@5.6" `but still the library mcrypt not installed

**Could you suggest any solution to this problem?**

Thanks
Bharat

**Comment by shivammathur** [comment, marked as answer]
For mcrypt extension, please use the [`shivammathur/extensions`](https://github.com/shivammathur/homebrew-extensions/) tap

```zsh
brew tap shivammathur/extensions
brew install mcrypt@5.6
```

**Reply by bharatkumarp** [reply]
Thanks a ton.  It solved the issue.

**Reply by ericknyoto** [reply]
> For mcrypt extension, please use the [`shivammathur/extensions`](https://github.com/shivammathur/homebrew-extensions/) tap
> 
> ```shell
> brew tap shivammathur/extensions
> brew install mcrypt@5.6
> ```

Is it also for another php extensions that need to install?
My composer json, need redis on php 7.3 to be installed. Maybe like this
```
brew tap shivammathur/extensions
brew install redis@7.3
```
